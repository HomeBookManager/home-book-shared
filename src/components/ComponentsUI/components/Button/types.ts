// types
import { TIconProps } from '../Icon/Icon';

export type TButtonIcon =
  | (Pick<TIconProps, 'iconComponent'> & {
      applyFill?: boolean;
      applyStroke?: boolean;
    })
  | null;

export type TButtonIconArgs = {
  placement: 'start' | 'end';
  src: TButtonIcon;
};
