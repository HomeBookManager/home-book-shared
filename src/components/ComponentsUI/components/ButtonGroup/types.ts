// types
import { TButtonProps } from '../Button/Button';

export type TButtonGroupPickedButtonProps = Pick<
  TButtonProps,
  | 'className'
  | 'color'
  | 'disabled'
  | 'disabledRippleEffect'
  | 'e2eAttribute'
  | 'e2eValue'
  | 'forcedHover'
  | 'size'
  | 'style'
  | 'type'
  | 'variant'
>;

export type TButtonGroupContext = TButtonGroupPickedButtonProps | undefined;
