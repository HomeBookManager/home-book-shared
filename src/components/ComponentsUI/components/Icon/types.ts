import {
  CSSProperties,
  HTMLAttributes,
  KeyboardEvent,
  MouseEvent,
} from 'react';

// types
import { IconShape } from './enums';

export type TIconBasicProps = HTMLAttributes<SVGSVGElement> & {
  className?: string;
  clickable?: boolean;
  disabled?: boolean;
  disabledFocus?: boolean;
  disabledHover?: boolean;
  e2eAttribute?: string;
  e2eValue?: number | string;
  fill?: string;
  forcedHover?: boolean;
  height?: string;
  onClick?: (event: MouseEvent) => void;
  onKeyDown?: (event: KeyboardEvent<SVGSVGElement>) => void;
  stroke?: string;
  style?: CSSProperties;
  tabIndex?: number;
  width?: string;
};

export type TIconShapes = { [key in IconShape]: string };
