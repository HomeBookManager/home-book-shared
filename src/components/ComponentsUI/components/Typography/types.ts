// @home-book
import {
  COMMON_COLORS,
  DARK_COLORS,
  FORM_COLORS,
  LIGHT_COLORS,
} from 'home-book-constants';
import { TValueObj, Theme } from 'home-book-types';

// types
import { TTypographyProps } from './Typography';

export type TTypographyColor = TValueObj<
  typeof COMMON_COLORS & typeof FORM_COLORS
>;

export type TTypographyColorMode = {
  [Theme.dark]: TValueObj<typeof DARK_COLORS>;
  [Theme.light]: TValueObj<typeof LIGHT_COLORS>;
};

export type TOmittedTypographyProps = Omit<TTypographyProps, 'fontType'>;
