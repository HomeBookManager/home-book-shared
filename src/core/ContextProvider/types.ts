// @home-book
import { Theme } from 'home-book-types';

export type TContext = {
  theme: Theme;
  setTheme: (theme: Theme) => void;
};
