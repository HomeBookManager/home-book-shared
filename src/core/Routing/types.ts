import { TObject } from 'home-book-types';
import {
  ComponentClass,
  ComponentType,
  FunctionComponent,
  LazyExoticComponent,
  ReactElement,
} from 'react';
import { RouteProps } from 'react-router';

export type TProtectedRouteProps = {
  guards?: Array<TGuard>;
  name: string;
} & RouteProps;

export type TGuard = {
  guardCheck: () => boolean;
  renderFallback: (
    routeProps: TProtectedRouteProps,
  ) => JSX.Element | ReactElement;
  translationKey: string;
};

export type TComponent =
  | ComponentType<{}>
  | LazyExoticComponent<FunctionComponent>
  | LazyExoticComponent<ComponentClass<{}, unknown>>
  | undefined;

export type TAppRouteData = {
  Component?: TComponent;
  guards?: Array<TGuard>;
  name: string;
  path?: string;
};

export type TAppRoutesData = Array<TAppRouteData | TAppRouteData['name']>;

export type TRoutes = TObject<string>;
