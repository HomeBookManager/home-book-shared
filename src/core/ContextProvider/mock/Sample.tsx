import { useContext } from 'react';

// @home-book
import { Theme } from 'home-book-types';

// core
import { Context } from '../ContextProvider';

const Sample = () => {
  const { theme, setTheme } = useContext(Context);

  return (
    <div>
      {theme}
      <button onClick={() => setTheme(Theme.dark)}>Click</button>
    </div>
  );
};

export default Sample;
