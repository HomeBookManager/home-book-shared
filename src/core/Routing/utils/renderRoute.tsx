// components
import ProtectedRoute from '../components/ProtectedRoute/ProtectedRoute';

// types
import { TAppRouteData, TRoutes } from '../types';

// utils
import { getRouteProps } from './getRouteProps';

export const renderRoute = (
  routeDataArg: TAppRouteData | TAppRouteData['name'],
  routes: TRoutes,
): JSX.Element => {
  const routeData: TAppRouteData =
    typeof routeDataArg === 'string' ? { name: routeDataArg } : routeDataArg;

  const routeProps = getRouteProps(routeData, routes);

  return <ProtectedRoute {...routeProps} key={routeData.name} />;
};
