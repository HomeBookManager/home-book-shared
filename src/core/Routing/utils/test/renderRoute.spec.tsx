import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';

// components
import Component from './mock/Component';

// core
import { renderRoute } from '../renderRoute';

// mocks
import { RouteName, routes } from '../../../../tests/mock/routing';

// types
import { TAppRouteData } from '../../types';

describe('renderRoute snaphots', () => {
  it('should render renderRoute', () => {
    // mock
    const routeData: TAppRouteData = {
      Component: Component,
      name: RouteName.home,
    };

    // before
    const { asFragment } = render(
      <Router>{renderRoute(routeData, routes)}</Router>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render when routeDataArg is string', () => {
    // mock
    const routeData: TAppRouteData['name'] = RouteName.home;

    // before
    const { asFragment } = render(
      <Router>{renderRoute(routeData, routes)}</Router>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
