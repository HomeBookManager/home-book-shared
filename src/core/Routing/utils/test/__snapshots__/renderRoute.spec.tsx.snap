// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderRoute snaphots should render renderRoute 1`] = `
<DocumentFragment>
  <div>
    Component
  </div>
</DocumentFragment>
`;

exports[`renderRoute snaphots should render when routeDataArg is string 1`] = `<DocumentFragment />`;
